serialization/php/json/src/JsonParseNode.php

```diff
@@ -113,19 +114,25 @@ public function getObjectValue(string $type): ?Parsable {
     }
 
     /**
-     * @param Parsable $result
+     * @param Parsable|AdditionalDataHolder $result
      * @return void
      */
-    private function assignFieldValues(Parsable $result): void {
-        $fieldDeserializers = $result->getFieldDeserializers();
+    private function assignFieldValues($result): void {
+        $fieldDeserializers = [];
+        if (is_a($result, Parsable::class)){
+            $fieldDeserializers = $result->getFieldDeserializers();
+        }
 
         foreach ($this->jsonNode as $key => $value){
             $deserializer = $fieldDeserializers[$key] ?? null;
 
             if ($deserializer !== null){
                 $deserializer($result, new JsonParseNode($value));
-            } else {
-                $result->getAdditionalData()[$key] = $value;
+            } else if (is_a($result, AdditionalDataHolder::class)) {
```

Comment:
testing whether it implements additionaldata holder and getting the additional data can be done outside of the loop to avoid paying the cost multiple times.
Then, assuming the array is returned by reference and not by copy (previous version of the code would suggest it), no need to set it each time

Reply:
Thanks for the feedback. Kindly have a look at the fix and let me know incase there are any issues.

Reply:
thanks, solving